OpenEdXMobile/src/main/java/org/edx/mobile/view/LearnFragment.kt

```diff
@@ -0,0 +1,132 @@
+        items = arrayListOf(LearnScreenItems.MY_COURSES)
+        if (environment.config.programConfig.isEnabled) {
+            items.add(LearnScreenItems.MY_PROGRAMS)
+            binding.llLearnSelection.setVisibility(true)
+            binding.llLearnSelection.setOnClickListener {
+                showLearnPopupMenu(binding.tvSelectedItem, items)
+            }
+        } else {
+            binding.llLearnSelection.setVisibility(false)
+        }
+        arguments?.let {
+            selectedItemPosition = it.getInt(SELECTED_POSITION, 0)
+        }
+        updateScreen(items[selectedItemPosition])
+    }
+
+    private fun showLearnPopupMenu(parentView: View, items: ArrayList<LearnScreenItems>) {
+        val listPopupWindow = ListPopupWindow(requireContext())
+        listPopupWindow.anchorView = parentView
+        val adapter = LearnDropDownAdapter(requireContext(), R.layout.learn_selection_item)
+        adapter.addAll(items)
+        listPopupWindow.setAdapter(adapter)
+        listPopupWindow.setOnItemClickListener { parent: AdapterView<*>?, _: View?, position: Int, _: Long ->
+            if (parent?.selectedItemPosition != position) {
+                selectedItemPosition = position
+                updateScreen(items[position])
+            }
+            listPopupWindow.dismiss()
+        }
+        listPopupWindow.show()
+        adapter.select(selectedItemPosition)
+        adapter.notifyDataSetChanged()
+        listPopupWindow.listView?.setSelector(android.R.color.transparent)
+    }
+
+    private fun updateScreen(item: LearnScreenItems) {
+        val screenName: String
+        val fragment: Fragment = when (item) {
+            LearnScreenItems.MY_COURSES -> {
+                screenName = Analytics.Screens.MY_COURSES
+                MyCoursesListFragment().apply {
+                    arguments = this@LearnFragment.arguments
+                }
+            }
+            LearnScreenItems.MY_PROGRAMS -> {
+                screenName = Analytics.Screens.MY_PROGRAM
+                WebViewProgramFragment.newInstance(environment.config.programConfig.url)
+            }
+        }
+        if (childFragmentManager.findFragmentByTag(fragment.javaClass.name) == null) {
+            val fragmentTransaction = childFragmentManager.beginTransaction()
+            fragmentTransaction.replace(binding.flLearn.id, fragment, fragment.javaClass.name)
+            fragmentTransaction.commit()
+        }
+        binding.tvSelectedItem.text = getString(item.labelRes)
+        environment.analyticsRegistry.trackScreenView(screenName)
+    }
+
+    @Subscribe
+    fun onEventMainThread(event: FragmentSelectionEvent) {
+        updateScreen(items[selectedItemPosition])
+    }
+
+    override fun isShowingFullScreenError(): Boolean {
+        return false
+    }
+
+    companion object {
+        private const val SELECTED_POSITION = "selected_position"
+    }
+
+    enum class LearnScreenItems(val labelRes: Int) {
```

Comment:
The name should be single i.e `LearnScreenItem`, no?

Reply:
done.